docs/versioned-plugins/inputs/logstash-v0.0.1.asciidoc

```diff
@@ -0,0 +1,252 @@
+
+[id="{version}-plugins-{type}s-{plugin}-minimum-config"]
+===== Minimum Configuration
+[cols="3a,2a"]
+|=======================================================================================================================
+|SSL Enabled              |SSL Disabled
+
+|
+
+[source,shell]
+----
+input {
+  logstash {
+    port => 8080
+    ssl_keystore_path
+         => "/path/to/logstash.p12"
+    ssl_keystore_password
+         => "${PASS}"
+  }
+}
+----
+
+|
+
+[source,shell]
+----
+input {
+  logstash {
+    port        => 8080
+    ssl_enabled => false
+  }
+}
+----
+
+|=======================================================================================================================
+
+
+[id="{version}-plugins-{type}s-{plugin}-config-binding"]
+===== Configuration Concepts
+
+This input plugin needs to be configured to bind to a TCP <<{version}-plugins-{type}s-{plugin}-port>>, and can be constrained to bind to a particular interface by providing the IP to <<{version}-plugins-{type}s-{plugin}-host>>.
+
+[id="{version}-plugins-{type}s-{plugin}-config-ssl-identity"]
+===== Security: SSL Identity
+
+Unless SSL is disabled, this plugin needs to be configured with identity material:
+
+* JKS- or PKCS12-formatted Keystore (see <<{version}-plugins-{type}s-{plugin}-ssl_keystore_path>>)
+* PKCS8-formatted Certificate/Key pair (see <<{version}-plugins-{type}s-{plugin}-ssl_certificate>>)
+
+[id="{version}-plugins-{type}s-{plugin}-config-ssl-trust"]
+===== Security: SSL Trust
+
+When communicating over SSL, this plugin can be configured to either request or require that connecting clients present their own identity claims with <<{version}-plugins-{type}s-{plugin}-ssl_client_authentication>>.
+
+Certificates that are presented by clients are validated by default using the system trust store to ensure that they are currently-valid and trusted, and that the client can prove possession of its associated private key.
+You can provide an _alternate_ source of trust with:
+
+* A PEM-formatted list of trusted certificate authorities (see <<{version}-plugins-{type}s-{plugin}-ssl_certificate_authorities>>)
+
+NOTE: Client-certificate verification does _not_ verify identity claims on the presented certificate, such as whether the certificate includes a Subject Alt Name matching the IP address from which the client is connecting.
+
+[id="{version}-plugins-{type}s-{plugin}-config-credentials"]
+===== Security: Credentials
+
+You can also configure this plugin to require a specific username/password be provided by configuring <<{version}-plugins-{type}s-{plugin}-username>> and <<{version}-plugins-{type}s-{plugin}-password>>.
+Doing so requires connecting {logstash-ref}/plugins-outputs-logstash.html[logstash output plugin] clients to provide matching <<plugins-outputs-{plugin}-username>> and <<plugins-outputs-{plugin}-password>>.
```

Comment:
Invalid reference probably because of these references. If so, similarly we have `plugins-inputs-logstash-username` references in output plugin docs 🙈 
```
14:54:55 INFO:build_docs:asciidoctor: WARNING: invalid reference: plugins-outputs-logstash-username
```

Reply:
Yes, I'm aware of what the issues are. I noted and anticipated them here: https://github.com/logstash-plugins/logstash-integration-logstash/pull/14. That's why I encourage everybody to avoid using links to output files before they are generated.